fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -230,33 +230,48 @@ void FixedLagSmoother::processQueue(fuse_core::Transaction& transaction)
   auto current_time = ros::Time(0, 0);
   if (!pending_transactions_.empty())
   {
-    current_time = pending_transactions_.front_key();
+    current_time = pending_transactions_.front().stamp();
   }
   // Attempt to process each pending transaction
-  while (!pending_transactions_.empty())
+  auto sensor_blacklist = std::vector<std::string>();
+  auto transaction_iter = pending_transactions_.rbegin();
+  while (transaction_iter != pending_transactions_.rend())
   {
-    auto& element = pending_transactions_.back_value();
-    // Apply the motion models to the transaction
-    if (!applyMotionModels(element.sensor_name, *element.transaction))
+    auto& element = *transaction_iter;
+    if (std::find(sensor_blacklist.begin(), sensor_blacklist.end(), element.sensor_name) != sensor_blacklist.end())
     {
+      // We should not process transactions from this sensor
+      ++transaction_iter;
+    }
+    else if (applyMotionModels(element.sensor_name, *element.transaction))
+    {
+      // Processing was successful. Add the results to the final transaction, delete this one, and move to the next.
+      transaction.merge(*element.transaction, true);
+      ++transaction_iter;
+      pending_transactions_.erase(transaction_iter.base());  // Reverse iterators are weird
+    }
+    else
+    {
+      // The motion model processing failed.
+      // Check the transaction timeout to determine if it should be removed or skipped.
       if (element.transaction->stamp() + params_.transaction_timeout < current_time)
       {
         // Warn that this transaction has expired, then skip it.
-        ROS_ERROR_STREAM("The queued transaction with timestamp " << element.transaction->stamp()
-                          << " could not be processed after " << (current_time - element.transaction->stamp())
-                          << " seconds, which is greater than the 'transaction_timeout' value of "
-                          << params_.transaction_timeout << ". Ignoring this transaction.");
-        pending_transactions_.pop_back();
-        continue;
+        ROS_ERROR_STREAM("The queued transaction with timestamp " << element.transaction->stamp() <<
+                          " from sensor " << element.sensor_name << " could not be processed after " <<
+                          (current_time - element.transaction->stamp()) << " seconds, which is greater " <<
+                          "than the 'transaction_timeout' value of " << params_.transaction_timeout <<
+                          ". Ignoring this transaction.");
+        ++transaction_iter;
```

Comment:
We have to use this twice. I wonder if a small free `reverse_erase` function would be valuable.